lib/withOnyx/index.tsx

```diff
@@ -272,26 +294,27 @@ export default function (mapOnyxToState, shouldDelayUpdates = false) {
             /**
              * Takes a single mapping and binds the state of the component to the store
              *
-             * @param {object} mapping
-             * @param {string|function} mapping.key key to connect to. can be a string or a
+             * @param mapping.key key to connect to. can be a string or a
              * function that takes this.props as an argument and returns a string
-             * @param {string} statePropertyName the name of the state property that Onyx will add the data to
-             * @param {boolean} [mapping.initWithStoredValues] If set to false, then no data will be prefilled into the
+             * @param statePropertyName the name of the state property that Onyx will add the data to
+             * @param [mapping.initWithStoredValues] If set to false, then no data will be prefilled into the
              *  component
-             * @param {string} key to connect to Onyx with
+             * @param key to connect to Onyx with
              */
-            connectMappingToOnyx(mapping, statePropertyName, key) {
+            connectMappingToOnyx(mapping: MapOnyxToState<TComponentProps, TOnyxProps>[keyof TOnyxProps], statePropertyName: keyof TOnyxProps, key: OnyxKey) {
+                const onyxMapping = mapOnyxToState[statePropertyName] as WithOnyxMapping<TComponentProps, TOnyxProps>;
+
                 // Remember what the previous key was so that key changes can be detected when data is being loaded from Onyx. This will allow
                 // dependent keys to finish loading their data.
                 // eslint-disable-next-line no-param-reassign
-                mapOnyxToState[statePropertyName].previousKey = key;
+                onyxMapping.previousKey = key;
 
                 // eslint-disable-next-line rulesdir/prefer-onyx-connect-in-libs
                 this.activeConnectionIDs[key] = Onyx.connect({
                     ...mapping,
                     key,
-                    statePropertyName,
-                    withOnyxInstance: this,
+                    statePropertyName: statePropertyName as string,
+                    withOnyxInstance: this as unknown as WithOnyxInstance,
```

Comment:
Does it work this way?
```suggestion
                    withOnyxInstance: this as WithOnyxInstance,
```

Reply:
Unfortunately no 🙁 [I spent some time on it yesterday but no luck so far](https://github.com/Expensify/react-native-onyx/pull/542#discussion_r1603824466)